cf_xarray/accessor.py

```diff
@@ -1157,23 +1122,64 @@ class CFAccessor:
     def __init__(self, obj):
         self._obj = obj
         self._all_cell_measures = None
+        self._flag_dict: Mapping[Hashable, FlagParam] | None = None
 
     def __setstate__(self, d):
         self.__dict__ = d
 
-    def _assert_valid_other_comparison(self, other):
-        # TODO cache this property
-        flag_dict = create_flag_dict(self._obj)
-        if other not in flag_dict:
+    @property
+    def flag_dict(self) -> Mapping[Hashable, FlagParam]:
+        """
+        Return possible flag meanings and associated bitmask/values.
+
+        The mapping values are a tuple containing a bitmask and a value. Either
+        can be None.
+        If only a bitmask: Independent flags.
+        If only a value: Mutually exclusive flags.
+        If both: Mix of independent and mutually exclusive flags.
+        """
+        if self._flag_dict is not None:
+            return self._flag_dict
```

Comment:
The problem is that you can change the underlying values in the DataArray and then the flags will be wrong.

Reply:
I forgot that if for DataArray.data most operations are not done in place (you get a new DataArray), changing `DataArray.attrs` is easily done in place. I only see either checking if any of the attributes have changed, or urging to use [`DataArray.assign_attrs()`](https://docs.xarray.dev/en/latest/generated/xarray.DataArray.assign_attrs.html) in the documentation...